File: linked-list/linked_list.f90
! linked_list.f90
program LinkedListExample
    implicit none
    type :: Node
        integer :: data
        type(Node), pointer :: next
    end type Node

    type(Node), pointer :: head => null()
    type(Node), pointer :: new_node, current

    call append(1)
    call append(2)
    call append(3)
    call append(4)
    call print_list()

contains

    subroutine append(data)
        integer, intent(in) :: data
        allocate(new_node)
        new_node%data = data
        new_node%next => null()
        if (.not. associated(head)) then
            head => new_node
        else
            current => head
            do while (associated(current%next))
                current => current%next
            end do
            current%next => new_node
        end if
    end subroutine append

    subroutine print_list()
        current => head
        do while (associated(current))
            print *, current%data
            current => current%next
        end do
    end subroutine print_list

end program LinkedListExample

File: linked-list/linked_list.cs
// linked_list.cs
using System;

class Node {
    public int data;
    public Node next;
    public Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    private Node head;

    public void Append(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
            return;
        }
        Node current = head;
        while (current.next != null) {
            current = current.next;
        }
        current.next = newNode;
    }

    public void PrintList() {
        Node current = head;
        while (current != null) {
            Console.Write(current.data + " ");
            current = current.next;
        }
        Console.WriteLine();
    }
}

class Program {
    static void Main() {
        LinkedList llist = new LinkedList();
        llist.Append(1);
        llist.Append(2);
        llist.Append(3);
        llist.Append(4);
        llist.PrintList();
    }
}
